.github/workflows/tf_changes.yaml

```
@@ -35,6 +35,72 @@ jobs:
         run: |
           echo "$ALL_CHANGED_FILES"
 
+  publish_model:
+    runs-on: ubuntu-latest
+    needs: tf_matrix
+    env:
+      TERRAFORM_DIR: "infrastructure/map"
+      # OCI CLI
+      OCI_CLI_USER: ${{ secrets.OCI_CLI_USER }}
+      OCI_CLI_TENANCY: ${{ secrets.OCI_CLI_TENANCY }}
+      OCI_CLI_FINGERPRINT: ${{ secrets.OCI_CLI_FINGERPRINT }}
+      OCI_CLI_KEY_CONTENT: ${{ secrets.OCI_CLI_KEY_CONTENT }}
+      OCI_CLI_REGION: ${{ secrets.OCI_CLI_REGION }}
+
+      # TF Backend Access Keys
+      AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+      AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+      TF_VAR_user_ocid: ${{ secrets.OCI_CLI_USER }}
+      TF_VAR_tenancy_ocid: ${{ secrets.OCI_CLI_TENANCY }}
+      TF_VAR_fingerprint: ${{ secrets.OCI_CLI_FINGERPRINT }}
+      TF_VAR_private_key: ${{ secrets.OCI_CLI_KEY_CONTENT }}
+      TF_VAR_region: ${{ secrets.OCI_CLI_REGION }}
+      TF_VAR_admin_password: ${{ secrets.OCI_DB_ADMIN_PASS }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Terraform Setup
+        uses: hashicorp/setup-terraform@v3.0.0
+        with:
+          terraform_version: 1.7.4
+
+      - name: Terraform Init
+        id: tf_init
+        working-directory: ${{ env.TERRAFORM_DIR }}
+        run: terraform init
+
+      - name: Terraform Apply
+        id: tf_apply
+        working-directory: ${{ env.TERRAFORM_DIR }}
+        run: |
+          if terraform apply -auto-approve 2>&1; then
+            echo "apply-outcome=success" >> $GITHUB_OUTPUT
+          else
+            echo "apply-outcome=failure" >> $GITHUB_OUTPUT
+          fi
+
+      - name: Terraform Output
+        id: tf_output
+        working-directory: ${{ env.TERRAFORM_DIR }}
+        run: |
+          if terraform output -json data > model.json 2>&1; then
```

this may swallow the error if `pipefail` is not set explicitly, I'm not sure.